### PR TITLE
test: use fgprof to show off-CPU profile in external package

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #48952

Problem Summary:

### What changed and how does it work?

also upgrade the ks3 SDK

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

we can see network IO is shown in CPU flame graph
image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/external/bench_test.go [hunk 1]
```diff
@@ -804,33 +843,67 @@ func TestReadMergeIterWithoutCheckHotspot(t *testing.T) {
...
+					maxInUse = m.HeapInuse
+					file, err2 := os.Create(filenameHeap)
+					intest.AssertNoError(err2)
+					err = pprof.WriteHeapProfile(file)
+					intest.AssertNoError(err)
+					err = file.Close()
+					intest.AssertNoError(err)
+				}
+			}
+		}()
```
Comment (ywqzzy) [inline br/pkg/lightning/backend/external/bench_test.go:900]: wrap a function?
Rest LGTM

Comment (lance6716) [reply br/pkg/lightning/backend/external/bench_test.go]: https://github.com/pingcap/tidb/pull/50179/commits/3bfa3b718896402abb2827062e122da3fdeadabc

**Review by ywqzzy — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Pull request merged**